### ISSUE TITLE
Database CPU pegged if user is a registrar user

### ISSUE THREAD

**Issue by rebeccacremona** [harvard-lil/perma#2886]
Why does this query use too much database CPU, if the user is a registrar user? Returns fast and causes no problems for org users or regular users; I did not test with admins. Totally independent of the number of links returned: problem happens for registrar with 1 link, and not for users with lots.

```
>>> queryset = Link.objects.order_by('-creation_timestamp').select_related('organization', 'organization__registrar', 'organization__shared_folder', 'capture_job', 'created_by').prefetch_related('captures').accessible_to(user)
>>> queryset.first()
```
https://github.com/harvard-lil/perma/blob/develop/perma_web/api/views.py#L338

**Comment by rebeccacremona** [comment]
This was untangled with help from Jack and Brennan and Ben back in January; see the [extended discussion in Slack](https://hlslil.slack.com/archives/C5U7T2B35/p1611686855007800) and https://github.com/harvard-lil/perma/pull/2887. 

TLDR, from Jack: 
> the query planner sees two options, which are “narrow down the list of links based on indexed lookups in the folder criteria, then sort them iteratively” or “get the sorted list of links with an index on creation_timestamp, then narrow them down iteratively based on the folder criteria.” We know that there aren’t that many links matching the folder criteria so the first one will be much faster, but the query planner is guessing otherwise (until we force it not to)

The theory is also that the query planner would have done better with this in a more up-to-date MySQL... or, that postgres would have done better... But in the meantime, clever use of subqueries forced the planner to do what we wanted.
